assets/wizards/advertising/views/placements/index.js

```diff
@@ -212,6 +213,24 @@ const Placements = ( { adUnits } ) => {
 								</>
 							);
 						} ) }
+					{ placement.supports?.indexOf( 'stick_to_top' ) > -1 && (
+						<CheckboxControl
+							label={ __( 'Stick to top', 'newspack' ) }
+							checked={ !! placement.data?.stick_to_top }
+							onChange={ value => {
+								setPlacements( {
+									...placements,
+									[ editingPlacement ]: {
+										...placements[ editingPlacement ],
+										data: {
+											...placements[ editingPlacement ].data,
+											stick_to_top: value,
+										},
```

Comment:
This deep object merging looks scary and error-prone. [lodash's `set`](https://lodash.com/docs/4.17.15#set) could be used to make it more readable and stable.

Reply:
Good call! 7c25a89d5a9159920f83bcc8266a6a650235c471 implements it with `lodash/fp` in respect to the immutability of the state variable.